### ISSUE TITLE
False-positive for S1871:Two branches in a switch-case when one branches use different overloads of methods

### ISSUE THREAD

**Issue by ikkentim** [SonarSource/sonar-dotnet#5587]
### Description

False positive when two branches in a switch-case use different overloads of the same method.

Rule:
S1871:Two branches in a conditional structure should not have exactly the same implementation

### Repro steps

```c#
Span<int> result;
switch (array)
{
    case int[] a:
        return new Span<int>(a, 0, length);
    case float[] a:
        result = new int[length];
        for (var i = 0; i < length; i++)
            result[i] = ValueConverter.ToInt32(a[i]);
        return result;
    case bool[] a: // S1871
        result = new int[length];
        for (var i = 0; i < length; i++)
            result[i] = ValueConverter.ToInt32(a[i]);
        return result;
    default:
        throw new InvalidOperationException("Unsupported array type");
}

// ...

public static class ValueConverter {
    public static int ToInt32(float f) => 0;
    public static int ToInt32(bool b) => 0;
}
```

### Expected behavior

No S1871

### Actual behavior

S1871

### Known workarounds

Workaround: Use a different name for one of the parameters in the case (e.g. `bool[] b`)

### Related information

- .NET 6, SDK 6.0.202 x64, latest C# version
- Visual Studio 2022
- `<PackageReference Include="SonarAnalyzer.CSharp" Version="8.37.0.45539">`
